src/Sparrow/Allocators.Pool.FixedSizeThreadAffine.cs

```diff
@@ -0,0 +1,329 @@
+{
+    public enum ThreadAffineWorkload : byte
+    {
+        Peaceful = 4,
+        Default = 16,
+        Contended = 64,
+        Absurd = 128
+    }
+
+    public interface IFixedSizeThreadAffinePoolOptions : IFixedSizeAllocatorOptions, IComposableAllocator<Pointer>
+    {        
+        bool AcceptOnlyBlocks { get; }
+        int ItemsPerLane { get; }
+        
+        ThreadAffineWorkload Workload { get; }
+    }
+
+    public static class FixedSizeThreadAffinePoolAllocator
+    {
+        public struct Default : IFixedSizeThreadAffinePoolOptions, INativeOptions
+        {
+            public bool UseSecureMemory => false;
+            public bool ElectricFenceEnabled => false;
+            public bool Zeroed => false;
+
+            public int BlockSize => 1 * Constants.Size.Megabyte;
+            public int ItemsPerLane => 4;
+
+            public bool AcceptOnlyBlocks => true;
+
+            public ThreadAffineWorkload Workload => ThreadAffineWorkload.Default;
+
+            public bool HasOwnership => true;
+            public IAllocatorComposer<Pointer> CreateAllocator()
+            {
+                var allocator = new Allocator<NativeAllocator<PoolAllocator.Default>>();
+                allocator.Initialize(default(PoolAllocator.Default));
+                return allocator;
+            }
+        }
+
+        public struct Static : IFixedSizeThreadAffinePoolOptions, INativeOptions, INativeGlobalAllocatorOptions
+        {
+            public bool UseSecureMemory => false;
+            public bool ElectricFenceEnabled => false;
+            public bool Zeroed => false;
+
+            public int BlockSize => 1 * Constants.Size.Megabyte;
+            public int ItemsPerLane => 4;
+
+            public bool AcceptOnlyBlocks => true;
+
+            public ThreadAffineWorkload Workload => ThreadAffineWorkload.Default;
+
+            public bool HasOwnership => true;
+            public IAllocatorComposer<Pointer> CreateAllocator()
+            {
+                var allocator = new Allocator<NativeAllocator<Static>>();
+                allocator.Initialize(default(Static));
+                return allocator;
+            }
+        }
+
+        public struct DefaultAcceptArbitratySize : IFixedSizeThreadAffinePoolOptions, INativeOptions
+        {
+            public bool UseSecureMemory => false;
+            public bool ElectricFenceEnabled => false;
+            public bool Zeroed => false;
+
+            public int BlockSize => 1 * Constants.Size.Megabyte;
+            public int ItemsPerLane => 4;
+
+            public bool AcceptOnlyBlocks => false;
+
+            public ThreadAffineWorkload Workload => ThreadAffineWorkload.Default;
+
+            public bool HasOwnership => true;
+            public IAllocatorComposer<Pointer> CreateAllocator()
+            {
+                var allocator = new Allocator<NativeAllocator<PoolAllocator.Default>>();
+                allocator.Initialize(default(PoolAllocator.Default));
```

Comment:
Shouldn't this be `DefaultAcceptArbitrarySize`?

Reply:
Yes, there is a bug in all 3.